apps/aecore/src/aec_fork_signalling.erl

```diff
@@ -0,0 +1,310 @@
+%%%============================================================================
+%%% @copyright (C) 2019, Aeternity Anstalt
+%%% @doc
+%%% TODO
+%%% @end
+%%%============================================================================
+-module(aec_fork_signalling).
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/0,
+         get_result/2,
+         compute_result/2
+        ]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2
+        ]).
+
+-export([worker_process/4]).
+
+-define(SERVER, ?MODULE).
+
+-type header_hash()      :: aec_blocks:block_header_hash().
+
+-type worker()           :: {pid(), reference()}.
+
+-type result()           :: true | false | pending.
+
+-type pointer()          :: header_hash().
+
+-record(state, {fork     :: aec_hard_forks:fork() | undefined,
+                results  :: #{header_hash() := result() | pointer()},
```

Comment:
A comment here on which header hashes are meant to be keys in this map would ease readability of intent.